### PR TITLE
Use feign client in `CcdClient`

### PR DESCRIPTION
**Pull request by rkondratowicz**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClient.java [hunk 1]
```diff
@@ -16,7 +22,33 @@ public CcdClient(CcdApi api, Supplier<String> s2sTokenSupplier) {
     }
     // endregion
 
-    public String createCase(Object data, String idamToken) {
-        throw new UnsupportedOperationException("Not implemented yet");
+    public String createCase(CaseCreationRequest req, TransformationResult tr) {
+        // TODO: handle exceptions
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClient.java]: Should it even handle exceptions at this level?

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClient.java [hunk 2]
```diff
@@ -16,7 +22,33 @@ public CcdClient(CcdApi api, Supplier<String> s2sTokenSupplier) {
...
+        StartEventResponse startEventResponse =
```
Comment (lgonczar) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClient.java:32]: Given two things are happening here, it's worth adding some logging after each step.

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClientTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        );
+
+        // when
+        String id = ccdClient.createCase(req, tr);
+
+        // then
+        assertThat(id).isEqualTo(newCaseId);
+
+        ArgumentCaptor<CaseDataReq> argCaptor = ArgumentCaptor.forClass(CaseDataReq.class);
+        verify(api).submitEvent(argCaptor.capture(), any(), any(), any(), any(), any(), anyBoolean());
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClientTest.java:81]: We could verify the exact values here.

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClientTest.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+    private CcdClient ccdClient;
+
+    @BeforeEach
+    public void setUp() {
+        ccdClient = new CcdClient(api, s2sTokenSupplier);
+    }
+
+    @Test
+    public void should_call_api_with_correct_params() {
```
Comment (lgonczar) [inline src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClientTest.java:41]: We can test if there will be no attempt to submit event when starting failed. Up to you.

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClient.java (hunk 2 above)
Comment (rkondratowicz) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClient.java:32]: done

File: src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClientTest.java (hunk 3 above)
Comment (rkondratowicz) [reply src/test/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/ccd/CcdClientTest.java:81]: it's implicit via mocking above.

**Review by lgonczar — APPROVED** [review]

**Pull request merged**